server/workschedulerserver.go

```diff
@@ -28,41 +26,25 @@ type WorkSchedulerServer struct {
 	pb.UnimplementedWorkSchedulerServer
 	Config     Config
 	grpcServer *grpc.Server
-	consumer   consumer.Consumer
+	workOrchestrator orchestrator.Orchestrator
 }
 
 // NewWorkSchedulerServer create a new WorkSchedulerServer
 func NewWorkSchedulerServer(c Config) *WorkSchedulerServer {
 	grpcServer := grpc.NewServer()
-	pb.RegisterWorkSchedulerServer(grpcServer, &WorkSchedulerServer{})
-
-	return &WorkSchedulerServer{
+	wsServer :=   &WorkSchedulerServer{
 		Config:     c,
 		grpcServer: grpcServer,
 	}
+
+	pb.RegisterWorkSchedulerServer(grpcServer, wsServer)
+	return wsServer
 }
 
 // Poll for work to process
 func (s *WorkSchedulerServer) Poll(ctx context.Context, in *empty.Empty) (*pb.Work, error) {
-	// blocks until there's work or until the context is terminated
-	msg, err := s.consumer.Poll(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	ws, err := json.Marshal(msg.TopicPartition)
-	if err != nil {
-		return nil, err
-	}
-
-	work := &pb.Work{
-		Source: &pb.WorkSource{
-			Source: string(ws),
-		},
-		Data:       msg.Value,
-	}
-
-	return work, nil
+	work := <-s.workOrchestrator.WorkChannel()
```

Comment:
I think that you want to respect the timeout of the ctx.

Reply:
Fixed